### PR TITLE
Docsite3

### PR DESCRIPTION
**Pull request by afoxman**
### Description

Another round of updates which add to or refine the documentation website.

- Introduction page
  - Add some context to the beginning
  - Re-organize and refine the rest of the page
- Dependencies page
  - Add meta capabilities
- Create a Getting Started guide
- Create a Dependency Management guide
- Use markdown fragments to share code/text between pages
- Styling updates
- Update the list of recommended VSode extensions
- Expand prettier formatting to include .mdx files

### CODE REVIEW

**Comment by afoxman** [comment]
@tido64 @kelset bump to get attention

**Comment by afoxman** [comment]
I know it's a pain (tell me if there's a better way) -- if you really want to see the content in this PR, it is most readable through the demo website. You'll have to clone my fork.

```sh
git clone https://github.com/afoxman/rnx-kit.git afoxman-rnx-kit
cd afoxman-rnx-kit
git checkout docsite3
yarn
cd docsite
yarn start
```

**Comment by kelset** [comment]
@afoxman thanks for the instructions, I was literally about to try it out 🤣

File: docsite/docs/guides/dependency-management.mdx [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+dependency manager.
+
+:::tip
+
+Not quite ready to make the leap? Learn more about
+[dependencies](/docs/dependencies) first.
+
+:::
+
+## On-boarding
```
Comment (kelset) [inline docsite/docs/guides/dependency-management.mdx:20]: noob question: is it "on-boarding" or "onboarding"?

File: docsite/docs/guides/dependency-management.mdx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+_Replace `yarn` with `npm run` or `pnpm`, depending on your package manager._
+
+When the validation process finds a compatibility problem, it fails with a
+detailed message, showing you what went wrong before you commit to the repo.
+
+:::tip Using a Task Runner?
+
+If you're using a task runner like [Lage](https://microsoft.github.io/lage) or
+[Gulp](https://gulpjs.com), you can go for a more sophistocated integration. The
```
Comment (kelset) [inline docsite/docs/guides/dependency-management.mdx:89]:
```suggestion
[Gulp](https://gulpjs.com), you can go for a more sophisticated integration. The
```
nit: typo

File: docsite/docs/guides/dependency-management.mdx [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+<PackageRunCommand args="rnx-dep-check --write" />
+
+**Review** and **test** the fixes before pushing them to the PR with the
+dependency manager update.
+
+## Upgrading React Native
+
+When you're ready to upgrade to a new release of React Native (or downgrade to
+an old release), you can use the dependency manager to do the heavy lifting.
```
Comment (kelset) [inline docsite/docs/guides/dependency-management.mdx:219]:
```suggestion
an old release), you can use the dependency manager to help you out.
```
One thing I think we need to make sure is to not confused anyone about the fact that folks will still have to do the hard work, which is doing all the changes the upgrade helper will tell them to do https://react-native-community.github.io/upgrade-helper/

we only change versions of stuff.

File: docsite/docs/guides/dependency-management.mdx [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+  `rnx-dep-check --vigilant [version] --write` command
+
+## Finish Line
+
+Congratulations! Your React Native apps and libraries are now up-to-date and
+compatible with each other! And they should stay that way thanks to the
+automation you've put in place!
+
+Was this guide helpful? Spread the word on [Twitter](https://twitter.com):
+`@ReactNativeMSFT` `#rnxkit`
```
Comment (kelset) [inline docsite/docs/guides/dependency-management.mdx:309]: I wonder if there's a way we can turn this into something similar to a callout (like the Tips or Alerts ones) and replicate it across the different pages

File: docsite/docs/introduction.md [hunk 5]
```diff
@@ -4,19 +4,39 @@ React Native engineering is complicated, and the ecosystem changes quickly. The
...
+that work well together and are compatble with each other? _And_ that match the
+React Native release you're using? There's a tool for that! It's all automated,
+too.
+
+Can't use Metro because it doesn't understand symlinks? Yarn, npm, and pmpm all
+use symlinks these days. There's a tool for that! A few lines of configuration,
+and you're ready to go.
+
+Do you use Metro and miss type-safe bundling and bundle-serving? There's a tool
+for that!
```
Comment (kelset) [inline docsite/docs/introduction.md:21]: personal preference: any time it says "there's a tool for that" I would want to be able to clink on [a tool for that]() and get sent directly to the page of it (either in the docs or the github repo)

File: docsite/docs/introduction.md [hunk 6]
```diff
@@ -27,25 +47,51 @@ them as overrides. Package configuration is _also_ optional, though it is
...
+// Run the dependency manager, scanning all packages and targeting React Native 0.66
 $ yarn react-native rnx-dep-check --vigilant 0.66
 ```
 
-### Á la carte
+### Choosing Individual Tools
+
+The tools are designed to be used individually. You can choose specific tools,
+and use them however and wherever they are needed. Mix and match with other
+tools and the CLI.
```
Comment (kelset) [inline docsite/docs/introduction.md:85]: maybe add a link/reference to the dedicated "tools" section?

**Review by kelset — APPROVED** [review]
thanks for all your work! I've left some comments but overall it's already 👍

File: docsite/docs/guides/dependency-management.mdx (hunk 1 above)
Comment (afoxman) [reply docsite/docs/guides/dependency-management.mdx:20]: Good catch! I checked a few online sources including the dictionary, and it should not be hyphenated. I fixed all of the references in this page.

File: docsite/docs/guides/dependency-management.mdx (hunk 4 above)
Comment (afoxman) [reply docsite/docs/guides/dependency-management.mdx:309]: I moved it into a markdown fragment file which we can embed anywhere like a UI control.

**Pull request merged**